src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,240 @@
+    else:
+        click.echo("No cases found.")
+
+
+def _display_file_events(sdk, case_number):
+    click.echo("\nFile Events:\n")
+    response = sdk.cases.file_events.get_all(case_number)
+    if not response["events"]:
+        click.echo("No events found.")
+    click.echo(pformat(json.loads(response.text)))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--include-file-events", is_flag=True, help="View events associated to the case."
+)
+@sdk_options()
+@format_option
+def show(state, case_number, format, include_file_events):
+    """Show case details."""
+    formatter = OutputFormatter(format, _get_cases_header())
+    try:
+        response = state.sdk.cases.get(case_number)
+        formatter.echo_formatted_list([response.data])
+        if include_file_events:
+            _display_file_events(state.sdk, case_number)
+    except Py42NotFoundError:
+        click.echo("Invalid case-number {}.".format(case_number))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--path", type=str, help="File path. Defaults to current directory.", default="."
+)
+@sdk_options()
+def export(state, case_number, path):
+    """Download case detail summary in a pdf file at the given path with name <case_number>_case_summary.pdf."""
+    response = state.sdk.cases.export_summary(case_number)
+    file = os.path.join(path, "{}_case_summary.pdf".format(case_number))
+    with open(file, "wb") as f:
+        f.write(response.content)
+
+
+@cases.group(cls=OrderedGroup)
+@sdk_options()
+def file_events(state):
+    """Fetch file events associated with the case."""
+    pass
+
+
+@file_events.command("list")
+@case_number_arg
+@sdk_options()
+@format_option
+def file_events_list(state, case_number, format):
+    """List all the events associated with the case."""
+    formatter = OutputFormatter(format, _get_events_header())
+    try:
+        response = state.sdk.cases.file_events.get_all(case_number)
+    except Py42NotFoundError as err:
+        click.echo("Invalid case-number.")
+        raise err
+    if not response["events"]:
+        click.echo("No events found.")
+    for event in response["events"]:
+        events = [event for event in response["events"]]
+        formatter.echo_formatted_list(events)
+
+
+@file_events.command()
+@case_number_arg
+@file_event_id_option
+@sdk_options()
+def add(state, case_number, event_id):
+    """Associate an event id to a case."""
+    try:
+        state.sdk.cases.file_events.add(case_number, event_id)
+    except Py42BadRequestError as err:
+        click.echo("Invalid case-number or event-id.")
+        raise err
+
+
+@file_events.command()
+@case_number_arg
+@file_event_id_option
+@sdk_options()
+def remove(state, case_number, event_id):
+    """Remove the associated event id from the case."""
+    try:
+        state.sdk.cases.file_events.delete(case_number, event_id)
+    except Py42NotFoundError as err:
+        click.echo("Invalid case-number or event-id.")
+        raise err
```

Comment:
Convert these two lines to just :  `raise Code42CLIError("Invalid case-number or event-id.")`


Two things here:

1.) In general, don't do `raise err`, just do `raise`, it preservers the stack for debuggin is easier.

2.) Instead of printing something and then raising an error, we have a process for raising CLI error so that they show up in red text and get a message printed

Reply:
Done.